x-pack/auditbeat/module/system/package/rpm_linux.go

```diff
@@ -238,65 +249,54 @@ func packageFromHeader(header C.Header, cFun *cFunctions) (*Package, error) {
 
 	pkg := Package{}
 
-	var name *C.char
-	res := C.my_headerGetEntry(cFun.headerGetEntry, header, RPMTAG_NAME, &name)
-	if res != 1 {
-		return nil, fmt.Errorf("Failed to call headerGetEntry(name): %d", res)
+	name := C.my_headerGetString(cFun.headerGetString, header, RPMTAG_NAME)
+	if name != nil {
+		pkg.Name = C.GoString(name)
+	} else {
+		return nil, errors.New("Failed to get package name")
 	}
-	pkg.Name = C.GoString(name)
 
-	var version *C.char
-	res = C.my_headerGetEntry(cFun.headerGetEntry, header, RPMTAG_VERSION, &version)
-	if res != 1 {
-		return nil, fmt.Errorf("Failed to call headerGetEntry(version): %d", res)
+	version := C.my_headerGetString(cFun.headerGetString, header, RPMTAG_VERSION)
+	if version != nil {
+		pkg.Version = C.GoString(version)
+	} else {
+		pkg.Error = errors.New("Failed to get package version")
 	}
-	pkg.Version = C.GoString(version)
 
-	var release *C.char
-	res = C.my_headerGetEntry(cFun.headerGetEntry, header, RPMTAG_RELEASE, &release)
-	if res != 1 {
-		return nil, fmt.Errorf("Failed to call headerGetEntry(release): %d", res)
+	release := C.my_headerGetString(cFun.headerGetString, header, RPMTAG_RELEASE)
+	if release != nil {
```

Comment:
Nit: You can simplify this code and most below by taking advantage of the fact that `C.GoString(nil)` returns an empty string, so there is no point on checking the return value and you can assign directly to the struct member.

`pkg.Release = C.GoString(C.my_headerGetString(...))`

Reply:
Nice, thanks! Let me put that in and make the code simpler.